iceoryx_utils/source/posix_wrapper/shared_memory_object/memory_map.cpp

```diff
@@ -123,20 +127,27 @@ MemoryMap& MemoryMap::operator=(MemoryMap&& rhs)
 
 MemoryMap::~MemoryMap()
 {
-    if (m_isInitialized)
-    {
-        cxx::makeSmartC(munmap, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {-1}, {}, m_baseAddress, m_length);
-    }
+    destroy();
 }
 
-bool MemoryMap::isInitialized() const
+bool MemoryMap::isInitialized() const noexcept
 {
     return m_isInitialized;
 }
 
-void* MemoryMap::getBaseAddress() const
+void* MemoryMap::getBaseAddress() const noexcept
 {
     return m_baseAddress;
 }
+
+void MemoryMap::destroy() noexcept
+{
+    if (m_isInitialized)
+    {
+        cxx::makeSmartC(munmap, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {-1}, {}, m_baseAddress, m_length);
```

Comment:
What happens if this call fails? Maybe return the error here and inform the user with `std:err` in the d'tor?